ivy/func_wrapper.py

```diff
@@ -45,6 +47,34 @@ def _get_first_array(*args, **kwargs):
 # ---------------#
 
 
+def handle_array_like(fn: Callable) -> Callable:
+
+    @functools.wraps(fn)
+    def new_fn(*args, **kwargs):
+        args = list(args)
+        parameters = list(typing.get_type_hints(fn))
+        annotations = list(typing.get_type_hints(fn).values())
+
+        # accessing common index across input arg values, parameters and annotations.
+        for i in range(len(annotations)):
```

Comment:
`for i, (annotation, parameter, arg) in enumerate(zip(annotations, parameters, args)):`